src/par_iter/string.rs

```diff
@@ -68,3 +81,103 @@ impl<'a> IntoIterator for ParChars<'a> {
         self.chars.chars()
     }
 }
+
+
+// /////////////////////////////////////////////////////////////////////////
+
+pub struct ParSplit<'a> {
+    chars: &'a str,
+    separator: char,
+
+    /// Keeps track of the first separator found in the string.  This lets us
+    /// quickly answer `can_split`, and it also corresponds to what parts we've
+    /// already scanned as we keep splitting smaller.
+    first: Option<usize>,
+}
+
+impl<'a> ParSplit<'a> {
+    fn new(chars: &'a str, separator: char) -> Self {
+        ParSplit {
+            chars: chars,
+            separator: separator,
+            first: chars.find(separator),
+        }
+    }
+}
+
+impl<'a> ParallelIterator for ParSplit<'a> {
+    type Item = &'a str;
+
+    fn drive_unindexed<C>(self, consumer: C) -> C::Result
+        where C: UnindexedConsumer<Self::Item>
+    {
+        bridge_unindexed(self, consumer)
+    }
+}
+
+impl<'a> UnindexedProducer for ParSplit<'a> {
+    fn can_split(&self) -> bool {
+        self.first.is_some()
+    }
+
+    fn split(self) -> (Self, Self) {
+        let ParSplit { chars, separator, first } = self;
+        let first = first.expect("error: splitting with no separator present!");
+
+        // First find a suitable UTF-8 boundary in the unsearched region.
+        let char_index = find_char_midpoint(&chars[first..]) + first;
+
+        // Find a separator in reverse, towards the `first` that we know exists.
+        let index = chars[first..char_index]
```

Comment:
I wonder if it is really worth the trouble to maintain the invariant that we know there exists a `first`? I initially imagined we would not have a field `first`. I guess the problem then is that we can't know whether `can_split` should return true? We could modify `can_split()` to take `&mut self`, so that it can update the producer's internal state (or perhaps remove it entirely, and instead have `split` return `Option<(Self, Self)>`). In that case, we could just do one search before the split, something like:

```rust
let char_index = find_char_midpoint(&chars);
let opt_index = chars[char_index..].find(separator).map(|i| i + char_index).or_else(|| /* search left */ );
...
```

It seems like a shame to have to do two searches sequentially before you can split, basically.